seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java

```diff
@@ -0,0 +1,94 @@
+import org.apache.seatunnel.engine.server.SeaTunnelServerStarter;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.instance.impl.HazelcastInstanceImpl;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Cluster fault tolerance test. Test the job recovery capability and data consistency assurance capability in case of cluster node failure
+ */
+public class ClusterFaultToleranceIT {
+
+    @BeforeAll
+    public static void beforeAll() {
+        TestUtils.initPluginDir();
+    }
+
+    @Test
+    public void testBatchJobRecoveryWhenWorkerDone() {
+        HazelcastInstanceImpl node1 =
+            SeaTunnelServerStarter.createHazelcastInstance(
+                TestUtils.getClusterName("ClusterFaultToleranceIT_testBatchJobRecoveryWhenWorkerDone"));
+
+        HazelcastInstanceImpl node2 =
+            SeaTunnelServerStarter.createHazelcastInstance(
+                TestUtils.getClusterName("ClusterFaultToleranceIT_testBatchJobRecoveryWhenWorkerDone"));
+
+        HazelcastInstanceImpl node3 =
+            SeaTunnelServerStarter.createHazelcastInstance(
+                TestUtils.getClusterName("ClusterFaultToleranceIT_testBatchJobRecoveryWhenWorkerDone"));
+
+        // waiting all node added to cluster
+        Awaitility.await().atMost(10000, TimeUnit.MILLISECONDS)
+            .untilAsserted(() -> Assertions.assertEquals(3, node1.getCluster().getMembers().size()));
+
+        // TODO Need FakeSource support parallel first
+        Common.setDeployMode(DeployMode.CLIENT);
+        String filePath = TestUtils.getResource("/test_cluster_fault_worker_batch_job.conf");
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setName("test_cluster_fault_worker_batch_job");
+
+        ClientConfig clientConfig = ConfigProvider.locateAndGetClientConfig();
+        clientConfig.setClusterName(
+            TestUtils.getClusterName("ClusterFaultToleranceIT_testBatchJobRecoveryWhenWorkerDone"));
+        SeaTunnelClient engineClient = new SeaTunnelClient(clientConfig);
+        JobExecutionEnvironment jobExecutionEnv = engineClient.createExecutionContext(filePath, jobConfig);
+        try {
+            final ClientJobProxy clientJobProxy = jobExecutionEnv.execute();
+
+            CompletableFuture<JobStatus> objectCompletableFuture = CompletableFuture.supplyAsync(() -> {
+                return clientJobProxy.waitForJobComplete();
+            });
+
+            Awaitility.await().atMost(20000, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> Assertions.assertTrue(
+                    objectCompletableFuture.isDone() && JobStatus.FINISHED.equals(objectCompletableFuture.get())));
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
The test will fail when the code in the unit test throws an exception. Therefore, there is no need to use try-catch to catch exceptions.

Reply:
done